runtime/jcl/common/mgmtthread.c

```diff
@@ -1287,8 +1292,28 @@ freeThreadInfos(J9VMThread *currentThread, ThreadInfo *allinfo, UDATA allinfolen
 		while (sinfo) {
 			next = sinfo->next;
 			j9mem_free_memory(sinfo);
+			if (NULL != sinfo->obj.unsafe) {
```

Comment:
Reference after free - free needs to move below the new code.

Reply:
oops! thanks @gacholio - fixes.